@NancyLi1013 I've removed the fail status for both linux and osx. I've also replaced the adhoc check on arch with vcpkg_fail_port_install, such that the build will fail with x86, arm and arm64.

The CI is properly showing the failures, as you explained. Now we know that these all pass when run via ParquetSharp CI.

**x64-linux**

This one really baffles me. It seems to go wrong in Arrow's FindBrotli.
```
-- Checking for modules 'libbrotlicommon;libbrotlienc;libbrotlidec'
--   Found libbrotlicommon, version 1.0.7
--   Found libbrotlienc, version 1.0.7
--   Found libbrotlidec, version 1.0.7
-- Configuring incomplete, errors occurred!
```
```
CMake Error at /mnt/_work/1/s/downloads/tools/cmake-3.17.2-linux/cmake-3.17.2-Linux-x86_64/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find Brotli (missing: BROTLI_COMMON_LIBRARY BROTLI_ENC_LIBRARY
  BROTLI_DEC_LIBRARY)
Call Stack (most recent call first):
  /mnt/_work/1/s/downloads/tools/cmake-3.17.2-linux/cmake-3.17.2-Linux-x86_64/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:445 (_FPHSA_FAILURE_MESSAGE)
  cmake_modules/FindBrotli.cmake:107 (find_package_handle_standard_args)
  /mnt/_work/1/s/scripts/buildsystems/vcpkg.cmake:329 (_find_package)
  cmake_modules/ThirdpartyToolchain.cmake:156 (find_package)
  cmake_modules/ThirdpartyToolchain.cmake:874 (resolve_dependency)
  CMakeLists.txt:458 (include)
```
However I cannot reproduce this error. It works absolutely fine when run on my machine and via GitHub Actions (see PR link in the description); note that both are running Ubuntu 18.04. This is what I get in my logs instead:
```
-- Checking for modules 'libbrotlicommon;libbrotlienc;libbrotlidec'
--   Found libbrotlicommon, version 1.0.7
--   Found libbrotlienc, version 1.0.7
--   Found libbrotlidec, version 1.0.7
-- Found Brotli: /home/gpsnoopy/Development/vcpkg.tanguyf/packages/brotli_x64-linux/debug/lib/libbrotlicommon-static.a
```
Not being able to reproduce the issue locally is making this incredibly difficult to figure out. Any ideas?

**x64-osx**

```
[1/95] cd /Volumes/data/work/1/s/buildtrees/thrift/src/401ebbec17-573d9ca574/compiler/cpp && /usr/bin/bison -d -o /Volumes/data/buildtrees/thrift/x64-osx-dbg/compiler/cpp/thrift/thrifty.cc /Volumes/data/work/1/s/buildtrees/thrift/src/401ebbec17-573d9ca574/compiler/cpp/src/thrift/thrifty.yy
FAILED: compiler/cpp/thrift/thrifty.cc compiler/cpp/thrift/thrifty.hh 
cd /Volumes/data/work/1/s/buildtrees/thrift/src/401ebbec17-573d9ca574/compiler/cpp && /usr/bin/bison -d -o /Volumes/data/buildtrees/thrift/x64-osx-dbg/compiler/cpp/thrift/thrifty.cc /Volumes/data/work/1/s/buildtrees/thrift/src/401ebbec17-573d9ca574/compiler/cpp/src/thrift/thrifty.yy
/Volumes/data/work/1/s/buildtrees/thrift/src/401ebbec17-573d9ca574/compiler/cpp/src/thrift/thrifty.yy:1.1-5: invalid directive: `%code'
/Volumes/data/work/1/s/buildtrees/thrift/src/401ebbec17-573d9ca574/compiler/cpp/src/thrift/thrifty.yy:1.7-14: syntax error, unexpected identifier
```

I believe the issue here is the Thrift dependency on Bison. The default version included in macOS is super old (2007 IIRC). This is why ParquetSharp CI does `brew install bison` on macOS before running vcpkg.

What's the best way to solve this? Sort of feels like this should be in the CI rather than in Thrift port.

--- a/ports/arrow/all.patch
+++ b/ports/arrow/all.patch
@@ -1,8 +1,8 @@
 diff --git a/cpp/cmake_modules/BuildUtils.cmake b/cpp/cmake_modules/BuildUtils.cmake
-index f5f0ad7..3dca82e 100644
+index e4e13cb70..58ca626da 100644
 --- a/cpp/cmake_modules/BuildUtils.cmake
 +++ b/cpp/cmake_modules/BuildUtils.cmake
-@@ -305,7 +305,7 @@ function(ADD_ARROW_LIB LIB_NAME)
+@@ -335,7 +335,7 @@ function(ADD_ARROW_LIB LIB_NAME)
        target_include_directories(${LIB_NAME}_static PRIVATE ${ARG_PRIVATE_INCLUDES})
      endif()
  
@@ -11,8 +11,22 @@ index f5f0ad7..3dca82e 100644
        set(LIB_NAME_STATIC ${LIB_NAME}_static)
      else()
        set(LIB_NAME_STATIC ${LIB_NAME})
+diff --git a/cpp/cmake_modules/FindBrotli.cmake b/cpp/cmake_modules/FindBrotli.cmake
+index bf47915c4..053e605a0 100644
+--- a/cpp/cmake_modules/FindBrotli.cmake
++++ b/cpp/cmake_modules/FindBrotli.cmake
+@@ -64,8 +64,7 @@ if(BROTLI_ROOT)
+             PATH_SUFFIXES ${INCLUDE_PATH_SUFFIXES}
+             NO_DEFAULT_PATH)
+ else()
+-  pkg_check_modules(BROTLI_PC libbrotlicommon libbrotlienc libbrotlidec)
+-  if(BROTLI_PC_FOUND)
++  if(0) # Find via pkg_check_modules disabled as incompatible with vcpkg
+     set(BROTLI_INCLUDE_DIR "${BROTLI_PC_libbrotlicommon_INCLUDEDIR}")
+ 
+     # Some systems (e.g. Fedora) don't fill Brotli_LIBRARY_DIRS, so add the other dirs here.
 diff --git a/cpp/cmake_modules/FindLz4.cmake b/cpp/cmake_modules/FindLz4.cmake
-index 8410916..a196b25 100644
+index 841091643..a196b251d 100644
 --- a/cpp/cmake_modules/FindLz4.cmake
 +++ b/cpp/cmake_modules/FindLz4.cmake
 @@ -19,14 +19,16 @@ if(MSVC AND NOT DEFINED LZ4_MSVC_STATIC_LIB_SUFFIX)
@@ -53,10 +67,10 @@ index 8410916..a196b25 100644
        PATH_SUFFIXES ${LIB_PATH_SUFFIXES})
      find_path(LZ4_INCLUDE_DIR NAMES lz4.h PATH_SUFFIXES ${INCLUDE_PATH_SUFFIXES})
 diff --git a/cpp/cmake_modules/FindThrift.cmake b/cpp/cmake_modules/FindThrift.cmake
-index f9d6296..82b8d22 100644
+index bb3eb5608..0b03d37d3 100644
 --- a/cpp/cmake_modules/FindThrift.cmake
 +++ b/cpp/cmake_modules/FindThrift.cmake
-@@ -54,6 +54,10 @@ if(MSVC AND NOT THRIFT_MSVC_STATIC_LIB_SUFFIX)
+@@ -43,6 +43,10 @@ if(MSVC AND NOT THRIFT_MSVC_STATIC_LIB_SUFFIX)
    set(THRIFT_MSVC_STATIC_LIB_SUFFIX md)
  endif()
  
@@ -67,7 +81,7 @@ index f9d6296..82b8d22 100644
  if(Thrift_ROOT)
    find_library(THRIFT_STATIC_LIB thrift${THRIFT_MSVC_STATIC_LIB_SUFFIX}
                 PATHS ${Thrift_ROOT}
-@@ -74,16 +78,14 @@ else()
+@@ -61,16 +65,14 @@ else()
  
      list(APPEND THRIFT_PC_LIBRARY_DIRS "${THRIFT_PC_LIBDIR}")
  
@@ -88,7 +102,7 @@ index f9d6296..82b8d22 100644
      find_path(THRIFT_INCLUDE_DIR thrift/Thrift.h PATH_SUFFIXES "include")
      find_program(THRIFT_COMPILER thrift PATH_SUFFIXES "bin")
 diff --git a/cpp/cmake_modules/FindZSTD.cmake b/cpp/cmake_modules/FindZSTD.cmake
-index 8e47086..d7ce559 100644
+index 8e47086e8..d87906a25 100644
 --- a/cpp/cmake_modules/FindZSTD.cmake
 +++ b/cpp/cmake_modules/FindZSTD.cmake
 @@ -19,14 +19,18 @@ if(MSVC AND NOT DEFINED ZSTD_MSVC_STATIC_LIB_SUFFIX)
@@ -112,7 +126,13 @@ index 8e47086..d7ce559 100644
                       "${CMAKE_SHARED_LIBRARY_PREFIX}zstd${CMAKE_SHARED_LIBRARY_SUFFIX}"
                 PATHS ${ZSTD_ROOT}
                 PATH_SUFFIXES ${LIB_PATH_SUFFIXES}
-@@ -44,14 +48,14 @@ else()
+@@ -39,19 +43,18 @@ if(ZSTD_ROOT)
+ 
+ else()
+   # Second, find via pkg_check_modules
+-  pkg_check_modules(ZSTD_PC libzstd)
+-  if(ZSTD_PC_FOUND)
++  if(0) # Disabled as incompatible with vcpkg
      set(ZSTD_INCLUDE_DIR "${ZSTD_PC_INCLUDEDIR}")
  
      list(APPEND ZSTD_PC_LIBRARY_DIRS "${ZSTD_PC_LIBDIR}")
@@ -130,10 +150,10 @@ index 8e47086..d7ce559 100644
                   PATH_SUFFIXES ${LIB_PATH_SUFFIXES})
      find_path(ZSTD_INCLUDE_DIR NAMES zstd.h PATH_SUFFIXES ${INCLUDE_PATH_SUFFIXES})
 diff --git a/cpp/cmake_modules/SetupCxxFlags.cmake b/cpp/cmake_modules/SetupCxxFlags.cmake
-index 75b33c2..80cac9a 100644
+index 6110a5aa5..3270d74a9 100644
 --- a/cpp/cmake_modules/SetupCxxFlags.cmake
 +++ b/cpp/cmake_modules/SetupCxxFlags.cmake
-@@ -128,7 +128,9 @@ macro(arrow_add_werror_if_debug)
+@@ -163,7 +163,9 @@ macro(arrow_add_werror_if_debug)
    if("${CMAKE_BUILD_TYPE}" STREQUAL "DEBUG")
      # Treat all compiler warnings as errors
      if(MSVC)

--- a/scripts/ci.baseline.txt
+++ b/scripts/ci.baseline.txt
@@ -1815,6 +1815,7 @@ theia:x64-windows-static = skip
 theia:x86-windows        = skip
 thor:x64-linux=fail
 thor:x64-osx=fail
+thrift:x64-osx=fail
 tidy-html5:arm-uwp=fail
 tidy-html5:x64-uwp=fail
 tinkerforge:arm-uwp=fail

